primary/src/proposer.rs

```diff
@@ -135,7 +135,23 @@ impl Proposer {
         self.last_parents = Certificate::genesis(&self.committee);
     }
 
-    // Main loop listening to incoming messages.
+    /// Compute the timeout value of the proposer.
+    fn timeout_value(&self) -> Instant {
+        match self.network_model {
+            // In partial synchrony, if this node is going to be the leader of the next
+            // round, we set a lower timeout value to increase its chance of committing
+            // the leader committed.
+            NetworkModel::PartiallySynchronous
+                if self.committee.leader(self.round + 1) == self.name =>
+            {
+                Instant::now() + self.max_header_delay / 2
```

Comment:
Why `max_header_delay / 2` ? Is there some logic why we picked to half the time or this is a good enough estimate for what we want to achieve?

Reply:
No deep reason (and open to suggestions). I just wanted to set a lower time than other nodes.